ci/eval/diff.nix

```diff
@@ -18,13 +18,22 @@ let
       added: [ <keys only in the second object> ],
       removed: [ <keys only in the first object> ],
       changed: [ <keys with different values between the two objects> ],
+      rebuilds: [ <keys in the second object with values not present at all in first object> ],
     }
   */
   diff =
+    old: new:
     let
       filterKeys = cond: attrs: lib.attrNames (lib.filterAttrs cond attrs);
+      oldOutputs = lib.pipe old [
+        (lib.mapAttrsToList (_: lib.attrValues))
+        lib.flatten
```

Comment:
Nit: I prefer to use `concatLists` (and/or `concatMap` as appropriate) when we only need to concatenate one level of lists.

`flatten` is (ofcourse) recursive, which can lead to unexpected behaviour if the input changes at some point.

Unless we actually need/want recursive concatenation for some reason?